File: c#/EmailMarketingApi/Locaweb.EmailMarketing.Api/Contatos/RepositorioContatos.cs
/*
 *  Copyright (c) 2009, Locaweb LTDA.
 * 	Todos os direitor reservados.
 *
 *  Está é uma API exemplo que facilita a utilização dos web services do Email Marketing.
 *
 * versao 0.1
 * mais detalhes em http://wiki.locaweb.com.br/pt-br/APIs_do_Email_Marketing
 */
using System;
using System.Collections.Generic;
using System.Net;

namespace Locaweb.EmailMarketing.Api.Contatos
{
    public class RepositorioContatos
    {
        /// <summary>
        /// Nome do servidor
        /// </summary>
        private string hostname;
        /// <summary>
        /// Login usado no Email Marketing
        /// </summary>
        private string login;
        /// <summary>
        /// Chave gerada para uso dessa API
        /// </summary>
        private string chave;

        private string hostnameSufix;
                
        private IEmktCore emktCore;

        public RepositorioContatos(string hostname, string login, string chave, string hostnameSufix)
        {
            this.hostname = hostname;
            this.login = login;
            this.chave = chave;
            this.hostnameSufix = hostnameSufix;
            this.emktCore = new EmktCore();
        }

        public RepositorioContatos(string hostname, string login, string chave, string hostnameSufix, IEmktCore emktCore)
        {
            this.hostname = hostname;
            this.login = login;
            this.chave = chave;
            this.hostnameSufix = hostnameSufix;
            this.emktCore = emktCore;
        }

        #region metodos publicos

        #region metodos de listagem de contatos
        
        public List<Contato> obterValidos(int pagina)
        {
            return this.obterPorStatus(pagina, "validos");
        }

        public List<Contato> obterInvalidos(int pagina)
        {
            return this.obterPorStatus(pagina, "invalidos");
        }

        public List<Contato> obterNaoConfirmados(int pagina)
        {
            return this.obterPorStatus(pagina, "nao_confirmados");
        }

        public List<Contato> obterDescadastrados(int pagina)
        {
            return this.obterPorStatus(pagina, "descadastrados");
        }
        #endregion

        #endregion

        #region metodos privados

        /// <summary>
        /// Retorna todos os contatos.         
        /// </summary>
        /// <param name="pagina">Número da página</param>
        /// <returns></returns>
        private List<Contato> obterPorStatus(int pagina, string status)
        {
            //nao deixa pagina negativa
            pagina = (pagina <= 0) ? 1 : pagina;

            string urlApi = string.Format("http://{0}.{1}/admin/api/{2}/contacts/{3}?chave={4}&pagina={5}",
                                       this.hostname,
                                       this.hostnameSufix,
                                       this.login,
                                       status,
                                       this.chave,
                                       pagina);


            string json = this.emktCore.GET(urlApi);
            return EmktCore.converteJsonParaObjeto<Contato>(json);

        }

        #endregion
    }
}

File: php/EmailMarketingApi/src/RepositorioContatos.php
<?php
require_once dirname(__FILE__).'/EmktCore.php';

/**
 *  Copyright (c) 2009, Locaweb LTDA.
 * 	Todos os direitor reservados.
 *
 *  Estс щ uma API exemplo que facilita a utilizaчуo dos web services do Email Marketing.
 *
 * @version 0.1
 * @see http://wiki.locaweb.com.br/pt-br/APIs_do_Email_Marketing
 */
class RepositorioContatos {

	private $HOSTNAME_SUFIX='locaweb.com.br';

	/**
	 * Nome do servidor.
	 */
  	private $hostName;

	/**
	 * Login usado no Email Marketing.
	 */
	private $login;

	/**
	 * Chave gerada para uso dessa API.
	 */
	private $chave;

	private $emktCore;

	private $hostNameSufix;

	/**
	 * @param string hostName usado no Email Marketing.
	 * @param string Login usado no Email Marketing.
	 * @param string Chave gerada para uso dessa API.
	 */
	public function RepositorioContatos($hostName, $login, $chave, $hostNameSufix='tecnologia.ws') {
		$this->hostName = $hostName;
		$this->login = $login;
		$this->chave = $chave;
		$this->hostNameSufix = $hostNameSufix;
		$this->emktCore = new EmktCore();
	}


/************************** Inicio metodos de Listagem de Contatos **************************************************
 * Os mщtodos de listagem possuem o parтmetro pagina. Ele informa qual pсgina da pesquisa deve ser retornada.
 * Atualmente o limite de contatos por pсgina щ de 25mil contatos por pсgina.
 * Por isso, caso tenha 40mil contatos em sua base por exemplo, precisarс fazer 2 chamadas passando
 * o parтmetro pagina=1 (que devolverс os contatos de 1 a 24999) e em seguida pagina=2 (que devolverс os contatos de 25000 a 40000)
 *
 */

	public function obterValidos($pagina='1'){
		return $this->obterPorStatus($pagina,'validos');
	}

	public function obterInvalidos($pagina='1'){
		return $this->obterPorStatus($pagina,'invalidos');
	}

	public function obterNaoConfirmados($pagina='1'){
		return $this->obterPorStatus($pagina,'nao_confirmados');
	}

	public function obterDescadastrados($pagina='1'){
		return $this->obterPorStatus($pagina,'descadastrados');
	}

	private function obterPorStatus($pagina='1', $status) {
		$url = "http://{$this->hostName}.{$this->hostNameSufix}/admin" .
				"/api/{$this->login}/contatos/{$status}?chave={$this->chave}&pagina={$pagina}";

		$resultado = $this->emktCore->enviaRequisicao($url);
		if($resultado==null) {
			return null;
		}
		$resultado = json_decode($resultado);
		if($resultado===null) {
			throw new Exception('Erro ao transformar em JSON.');
		}

		return $resultado;
	}

	public function setEmktCore($emktCore) {
		$this->emktCore = $emktCore;
	}
}
?>